ex2_system/source/logger/logger.c

```diff
@@ -63,6 +66,12 @@ static void do_output(const char *str) {
 
     uint32_t uptime = (uint32_t)(xTaskGetTickCount()/configTICK_RATE_HZ);
 
+    if (uptime > next_swap) {
+        stop_logger_fs(); // reset the logger file
+        init_logger_fs();
+        next_swap += LOGGER_SWAP_PERIOD_MS; // if this overflows that is okay. The tick count also overflows;

```

Comment:
@wagnerd97  TODO: Take over this PR and refactor this to go by # logs, not time.